Persistent attributes of a Panache entity should not be generated when using Jackson

the-bug opened the pull request:
When using JSONB the the method *isPersistent()* in PanacheEntityBase is ignored from JSONB and does not produce the *persistant*-attribute. This was fixed in https://github.com/FroMage/quarkus/commit/86edaa10eba0da2c1052566d752a257068da2b0f
But if Jackson is used, the *persistant*-attribute is still generated. 

This PR tackles this issue.

This is my first PR in an open source project and I do not know if i mess things up....

cescoffier commented:
@FroMage can you have a look?

FroMage commented on integration-tests/hibernate-orm-panache/jackson-integration/src/test/java/io/quarkus/it/panache/TestEndpointTest.java:
```diff
@@ -0,0 +1,26 @@
+                .then() //
+                .statusCode(200) //
+                // here is checked that the property 'persistent' is missing
+                .body(is("{\"id\":null,\"name\":\"max\"}"));
```
This doesn't seem right: the body should be `{"name": "max"}`.

the-bug replied:
> Hi. Thanks for the PR. The fix looks good, but I don't think we should add a new integration test just for this. Can't we add a unit test to the existing integration test using Jackson explicitely instead?

Hi,
thanks for the review!
I have build a new integration test module for this because I do not know how to configure Quarkus to use JSONB and Jackson together... I was following this guide: https://quarkus.io/guides/rest-json 
How can I tell Quarkus to use JSONB or Jackson for a resource or method explicitly?

> This doesn't seem right: the body should be `{"name": "max"}`.

Yes, this is not right.
The behaviour with jsonb is different. There it is `{"name": "max"}` with no `"id":null`.
In PanacheEntity is mentioned that the accessors for id are generated but I do not know how...
I do not know to fix this. Do you have a hint?

FroMage replied:
Don't use REST, just write a unit test that serialises using the Jackson API?

Apologies, my bad, I misread and I think it's right that if the `id` is null it can be serialised as such. I initially thought it was `persistent` instead of `id`. Just ignore this :)

the-bug replied:
> Don't use REST, just write a unit test that serialises using the Jackson API?

Thank you for that hint. Did it.

FroMage reviewed (changes requested):
Hi. Thanks for the PR. The fix looks good, but I don't think we should add a new integration test just for this. Can't we add a unit test to the existing integration test using Jackson explicitely instead?

Commit:
isPersistent is ignored from Jackson

Jackson did use the isPersistent() method to generate a 'persistent'
attribute in the JSON-body. This attribute has no meaning for clients
consuming this JSON-object.
With the annotation @JsonIgnore the attribute is not generated anymore.

FroMage reviewed (changes requested):
OK, thanks, this is much better. Can you please remove the formatting changes of the `pom.xml` files? This  doesn't let us view the changes, and I don't think they're necessary.

the-bug commented:
Sorry, I did not see I messed with the formatting...

FroMage reviewed (approved):
Perfect, thanks!

FroMage commented:
One CI test failed, looks unrelated, rerunning it.

the-bug commented:
Can the PR be merged?

FroMage commented:
Absolutely, thanks!
